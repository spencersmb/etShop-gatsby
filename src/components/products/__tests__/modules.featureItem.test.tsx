import FeatureItem from '@components/products/modules/features/featureItem'
import React from 'react'
import renderer from 'react-test-renderer'
import {
	render,
	cleanup
} from 'react-testing-library'
import 'jest-dom/extend-expect'

afterEach(cleanup)

const propsDefault = {
	description: 'description',
	icon: 'texture',
	title: 'title 1',
	index: 1
}
describe('Feature Item Test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(
				<FeatureItem {...propsDefault}/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('Should render correct title', () => {
		const modalRender = render(<FeatureItem {...propsDefault}/>)
		expect(modalRender.getByTestId('title').innerHTML).toEqual(propsDefault.title)
	})
	it('Should render correct description', () => {
		const modalRender = render(<FeatureItem {...propsDefault}/>)
		expect(modalRender.getByTestId('desc').innerHTML).toEqual(propsDefault.description)
	})
})
