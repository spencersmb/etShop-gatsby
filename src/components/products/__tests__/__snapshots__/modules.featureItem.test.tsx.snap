// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Description renders correctly 1`] = `
<div
  className="sc-bxivhb fyrUIa"
>
  <div
    className="sc-htpNat gYWYhm"
    color="#18C3C3"
    data-testid="title"
  >
    title 1
  </div>
  <div
    className="sc-bwzfXH ihTQNy"
    data-testid="desc"
  >
    description
  </div>
  <div
    className="sc-bdVaJa hkMwfE"
    color="#18C3C3"
  >
    <test-file-stub />
  </div>
</div>
`;
